### ISSUE TITLE
[BUG] `use-setstate-synchronously` with switch cases and exception on handling.

### ISSUE THREAD

**Issue by asaarnak** [dart-code-checker/dart-code-metrics#1133]
See comments in code examples

When using exception on handling.
```dart
} on FirstException {
    if (!mounted) {
      return;
    }
    setState(() {
    });
} on SecondException {
    // Missing analyzer warning here
    setState(() {
    });
 }
```
When using switch case handling.
```dart
    switch (enumStatus) {
      case MyEnum.FIRST:
        if (!mounted) {
          return;
        }
        setState(() {
        });
        return;
      case MyEnum.SECOND:
        // Missing analyzer warning here
        setState(() {
        });
        return;
    }
```

**Comment by incendial** [comment]
@Desdaemon hi, do you want to take a look?

**Comment by incendial** [comment]
Fixed in 5.5.0 🚀
